### PR TITLE
Add big doc to wrap_fx_proxy_cls

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103407

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78

### CODE REVIEW

File: torch/_dynamo/variables/builder.py [hunk 1]
```diff
@@ -1072,6 +1072,44 @@ def wrap_fx_proxy(tx, proxy, example_value=None, **options):
 
 # Note: Unfortunate split due to some gross classes existing that subclass TensorVariable
 # Should be compositional instead
+#
+# This is a horribly complicated function that does too many things, to
+# explain what it does, let's first talk about the classic usage wrap_fx_proxy
+# for a TensorVariable.  There are two primary modes of use:
```
Comment (voznesenskym) [inline torch/_dynamo/variables/builder.py:1078]: Do you want to move this part up to `wrap_fx_proxy`?

File: torch/_dynamo/variables/builder.py [hunk 2]
```diff
@@ -1072,6 +1072,44 @@ def wrap_fx_proxy(tx, proxy, example_value=None, **options):
...
+#     tracing but don't otherwise return anything.
+#
+#   * Some operators return symbolic ints/floats/bools which can go in the
+#     graph and be traced (but only if they're actually symbolic!  If they're
+#     static you don't want to put them in the graph, which means you
+#     shouldn't call this function.)
+#
+# The common theme is that you only use this function WHEN YOU ARE TRACING
+# SOMETHING INTO THE GRAPH.  This is sort of obvious, because you can't call
+# this function without a proxy.
```
Comment (voznesenskym) [inline torch/_dynamo/variables/builder.py:1112]: This used to be called TensorVariable.create

wrap_fx_proxy was a contract I added to force people to think about "What is my proxy to be?"

I think the next iteration should be something like `proxy.to_graph_as()`

File: torch/_dynamo/variables/builder.py (hunk 1 above)
Comment (ezyang) [reply torch/_dynamo/variables/builder.py:1078]: No? Comment should be close to the relevant implementation.

**Review by voznesenskym — APPROVED** [review]

**Pull request closed without merging**